src/libraries/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs

```diff
@@ -1629,18 +1651,18 @@ private static void AssertHeadersAreInjected(HttpRequestData request, Activity p
                 Assert.StartsWith($"00-{parent.TraceId.ToHexString()}-", traceparent);
                 Assert.Equal(passthrough, parent.Id == traceparent);
                 Assert.Equal(parent.TraceStateString, tracestate);
-            }
 
-            List<NameValueHeaderValue> correlationContext = (GetHeaderValue(request, "Correlation-Context") ?? string.Empty)
-                .Split(',', StringSplitOptions.RemoveEmptyEntries)
-                .Select(kvp => NameValueHeaderValue.Parse(kvp))
-                .ToList();
+                List<NameValueHeaderValue> correlationContext = (GetHeaderValue(request, "baggage") ?? string.Empty)
+                    .Split(',', StringSplitOptions.RemoveEmptyEntries)
+                    .Select(kvp => NameValueHeaderValue.Parse(kvp))
+                    .ToList();
 
-            List<KeyValuePair<string, string>> baggage = parent.Baggage.ToList();
-            Assert.Equal(baggage.Count, correlationContext.Count);
-            foreach (var kvp in baggage)
-            {
-                Assert.Contains(new NameValueHeaderValue(kvp.Key, kvp.Value), correlationContext);
+                List<KeyValuePair<string, string>> baggage = parent.Baggage.ToList();
+                Assert.Equal(baggage.Count, correlationContext.Count);
+                foreach (var kvp in baggage)
+                {
+                    Assert.Contains(new NameValueHeaderValue(kvp.Key, kvp.Value), correlationContext);
+                }
```

Comment:
Would it make sense to keep this check working for `ActivityIdFormat.Hierarchical`? Isn't it possible to query either `baggage`  or `Correlation-Context` depending on the values of `isW3C` and `parent.IdFormat`.

Reply:
The new W3C propagator doesn't support activities created with `ActivityIdFormat.Hierarchical` so the baggage will not be injected to extract it.

```C#
        public override void Inject(Activity? activity, object? carrier, PropagatorSetterCallback? setter)
        {
            if (activity is null || setter is null || activity.IdFormat != ActivityIdFormat.W3C)
            {
                return;
            }
```

Reply:
What I meant is that in (at leaset some of) the tests using `AssertHeadersAreInjected`, we could set the `LegacyPropagator` in case `idFormat = ActivityIdFormat.Hierarchical` and pass `isW3C = false` to this method, so it can keep testing the `LegacyPropagator` behavior.

Note that this is nitpicking and we may decide to not care about keeping the coverage for `LegacyPropagator`. I leave the decision to you.

Reply:
This is already covered in the `SendAsync_CustomSocketsHttpHandlerPropagator_PropagatorIsUsed(DistributedContextPropagator propagator, ActivityIdFormat idFormat)` which one of the cases will call `AssertHeadersAreInjected` with `isW3C = false` and `ActivityIdFormat.Hierarchical`. You may look at `SocketsHttpHandlerPropagators_WithIdFormat_MemberData` for the combinations of the propagators and ID formats.